index.html

```diff
@@ -16,57 +16,56 @@ <h1>
 		</h1>
 	</header>
 
-	<fieldset>
-		Resolution:
-		<input id="width" inputmode="numeric" pattern="\d+" />
-		×
-		<input id="height" inputmode="numeric" pattern="\d+" />
-
-		<p id="resolutions">Common:
-			<a href="#" tabindex="-1">3840×2160</a>
-			<a href="#" tabindex="-1">2560×1440</a>
-			<a href="#" tabindex="-1">1920×1080</a>
-			<a href="#" tabindex="-1">1680×1050</a>
-			<a href="#" tabindex="-1">1440×900</a>
-			<a href="#" tabindex="-1">1366×768</a>
-			<a href="#" tabindex="-1">1280×800</a>
-			<a href="#" tabindex="-1">1024×768</a>
-		</p>
-	</fieldset>
-
-	<fieldset>
-		<select id="dimension" title="Physical dimension (in inches)">
-			<option selected value="d">Diagonal</option>
-			<option value="w">Width</option>
-			<option value="h">Height</option>
-		</select>:
-		<input id="physical" value="13.3" inputmode="numeric" pattern="\d*\.?\d+" />″
-
-		<p id="diagonals">Common diagonals:
-			<a href="#" tabindex="-1">7″</a>
-			<a href="#" tabindex="-1">11.6″</a>
-			<a href="#" tabindex="-1">13.3″</a>
-			<a href="#" tabindex="-1">14″</a>
-			<a href="#" tabindex="-1">15.6″</a>
-			<a href="#" tabindex="-1">17.3″</a>
-			<a href="#" tabindex="-1">21″</a>
-			<a href="#" tabindex="-1">24″</a>
-			<a href="#" tabindex="-1">27″</a>
-		</p>
-	</fieldset>
-
-	<output>
-		<div>
-			<strong id="result"></strong>
-			pixels per inch
-		</div>
-	</output>
+	<section id="dpi-calc" class="mv-autoedit" mv-app mv-storage="https://github.com/LeaVerou/dpi/blob/gh-pages/devices.json" mv-bar="no-login">
+		<fieldset>
+			Resolution:
+			<input id="width" property inputmode="numeric" pattern="\d+" class="mv-autosize" mv-storage="none">
+			×
+			<input id="height" property inputmode="numeric" pattern="\d+" class="mv-autosize" mv-storage="none">
+			<meta property="ratio" content="[width/height]">
+			<p id="resolutions">Common:
+				<a mv-multiple="resolution" href="#[w]×[h]" tabindex="-1" mv-action="set(width, w) set(height, h)" mv-attribute="none">
+					<span property="w"></span>×<span property="h"></span>
+				</a>
+				<a href="" class="mv-logged-out"><strong class="mv-login">Add</strong></a>
```

Comment:
I haven't looked at this live, but I suspect the default UI may be a little jarring, we may want to hide `.mv-add-resolution` and `mv-add-diagonal` in the CSS.

Reply:
What do you mean by "hide `.mv-add-resolution` and `mv-add-diagonal` in the CSS"? Could you elaborate on that?

Reply:
These are classes to target the Add buttons that Mavo generates.